pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -87,10 +88,18 @@ public MultiStageBrokerRequestHandler(PinotConfiguration config, BrokerRoutingMa
     }
     _reducerHostname = reducerHostname;
     _reducerPort = config.getProperty(QueryConfig.KEY_OF_QUERY_RUNNER_PORT, QueryConfig.DEFAULT_QUERY_RUNNER_PORT);
+    long brokerTimeoutMs = config.getProperty(CommonConstants.Broker.CONFIG_OF_BROKER_TIMEOUT_MS, -1L);
```

Comment:
Should we handle the case when broker timeout is configured to be less than the query runner timeout? If someone only sets the broker timeout as 5s and uses a default query-runner timeout (10s), Mailbox operators may dangle for an additional 5s even though broker would have already hung up.

Reply:
> If someone only sets the broker timeout as 5s and uses a default query-runner timeout (10s)

I don't think the code should allow this. From user's perspective, either timeOut is overridden or not (default will be used). 

If someone has overridden then we should not be using the default limits at any level. User should only be allowed to override at the coarse / higher level as that is what matters to them.

The delta of (overridden value - time spent in planning on broker) should be used and propagated down to other stages